File: kotlin/hamming/src/main/kotlin/Hamming.kt
class Hamming {
    companion object {

        fun compute(a: String, b: String): Int {
            require(a.length == b.length) { "left and right strands must be of equal length." }
            return a.indices.count { a[it] != b[it] }
        }

        // works, too
        fun compute3(a: String, b: String): Int {
            require(a.length == b.length) { "left and right strands must be of equal length." }
            return a.filterIndexed { i, c -> b[i] == c }.count()
        }

        // works but I find the !it not very readable
        fun compute2(a: String, b: String): Int {
            require(a.length == b.length) { "left and right strands must be of equal length." }
            return a.mapIndexed { i, c -> b[i] == c }.count { !it }
        }
    }
}

File: go/hamming/hamming.go
package hamming

import (
	"errors"
)

// Distance of two strings measured in count of non matching chars
func Distance(a, b string) (out int, err error) {
	if len(a) != len(b) {
		return 0, errors.New("left and right strands must be of equal length")
	}
	for i := range a {
		if a[i] != b[i] {
			out++
		}
	}
	return out, nil
}

/*
fun compute(a: String, b: String): Int {
	require(a.length == b.length) { "left and right strands must be of equal length." }
	return a.indices.count { a[it] != b[it] }
}
*/
